### PR TITLE
Fix s3-style link mapper for gcs

### LINKED ISSUE #62193: Google Storage s3-style(gs://) URL doesn't work
Version 24.3.1: `gs://bucket/data` link transforms to `https://bucket.storage.googleapis.com/data`, but it doesn't work
```
d97f8c6a4ac5 :) select PRODUCT_ID from gcs('gs://gcp-public-data-sentinel-2/L2/index.csv.gz') limit 1

SELECT PRODUCT_ID
FROM gcs('gs://gcp-public-data-sentinel-2/L2/index.csv.gz')
LIMIT 1

Query id: 60bbc9a0-4706-4843-ba40-fc43f154bfec


Elapsed: 0.017 sec. 

Received exception from server (version 24.3.1):
Code: 36. DB::Exception: Received from localhost:9031. DB::Exception: Bucket name length is out of bounds in virtual hosted style S3 URI: 'L2' (https://gcp-public-data-sentinel-2.storage.googleapis.com/L2/index.csv.gz). (BAD_ARGUMENTS)
```
it should be transformed to `https://storage.googleapis.com/bucket/data`
```
d97f8c6a4ac5 :) select PRODUCT_ID from gcs('https://storage.googleapis.com/gcp-public-data-sentinel-2/L2/index.csv.gz') limit 1

SELECT PRODUCT_ID
FROM gcs('https://storage.googleapis.com/gcp-public-data-sentinel-2/L2/index.csv.gz')
LIMIT 1

Query id: 42ae7a2c-1451-4726-8801-688952df4bf5

   ┌─PRODUCT_ID───────────────────────────────────────────────────┐
1. │ S2A_MSIL2A_20230805T044701_N0509_R076_T45SYS_20230805T084102 │
   └──────────────────────────────────────────────────────────────┘

1 row in set. Elapsed: 0.469 sec. 
```

### PR DESCRIPTION
**Pull request by yariks5s**
Changed mapper for gcs from `https://{bucket}.storage.googleapis.com/{path}` to `https://storage.googleapis.com/{bucket}/{path}`. Fixes #62193

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [x]  Performance tests
- [x]  All with ASAN
- [x]  All with TSAN
- [x]  All with MSAN
- [x]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit ca27cf3fde37442f76f3d244f8cc57b5c541ebc7 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62257/ca27cf3fde37442f76f3d244f8cc57b5c541ebc7/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: programs/server/config.xml [hunk 1]
```diff
@@ -96,7 +96,7 @@
             <to>https://{bucket}.s3.amazonaws.com</to>
         </s3>
         <gs>
-            <to>https://{bucket}.storage.googleapis.com</to>
+            <to>https://storage.googleapis.com/{bucket}</to>
```
Comment (antonio2368) [inline programs/server/config.xml:99]: So this was a mistake and `https://{bucket}.storage.googleapis.com` is always wrong?

**Review by antonio2368 — COMMENTED** [review]
LGTM

As the issue can be fixed with a config change, should we recommend the user to do it until the fix lands?

File: programs/server/config.xml (hunk 1 above)
Comment (yariks5s) [reply programs/server/config.xml:99]: yes, `https://{bucket}.storage.googleapis.com` is always wrong

> As the issue can be fixed with a config change, should we recommend the user to do it until the fix lands?

- user already has a solution for this problem, as he uses a direct link instead s3-style link

Comment (antonio2368) [reply programs/server/config.xml:99]: I think it still makes sense to make them aware that they can modify config if they still want to use s3-style links.

**Review by antonio2368 — APPROVED** [review]

**Comment by yariks5s** [comment]
Builds with sanitizers were not found because I disabled all tests with sanitizers

**Pull request merged**